src/cljs/rems/language.cljs

```diff
@@ -31,13 +42,12 @@
 
 (defn update-language [language]
   (set! (.. js/document -documentElement -lang) (name language))
+  (set-language-cookie! language)
+  (rf/dispatch [:rems.flash-message/reset]) ; may have saved localized content
   (update-css language))
 
 (reg-event-fx
  ::set-language
- (fn [{:keys [db]} [_ language]]
-   (let [user-id (get-in db [:identity :user :eppn])]
-     (if user-id
-       (rf/dispatch [:rems.user-settings/update-user-settings {:language language}])
-       (do (update-language language)
-           {:db (assoc db :non-user-language language)})))))
+ (fn [_ [_ language]]
+   (update-language language)
+   {:dispatch [:rems.user-settings/update-user-settings {:language language}]}))
```

Comment:
if we merged with existing settings here, we wouldn't need to merge in update-user-settings

Reply:
It is likely that the settings will be a complicated map, so it's more useful to have the caller do less work than to remember to set everything? There used to be no merge which would eventually lead to bugs.

Reply:
Maybe even a deep-merge here? I agree a plain setter is "simpler" in some sense.

Reply:
I think I would be fine with having deep-merge in `update-user-settings`, but splitting the PUT to a different event. See my comment on `update-user-settings`.